### ISSUE TITLE
image-cache fails when no image at url

### ISSUE THREAD

**Issue by JWiseCoder** [NativeScript/NativeScript#7033]
When using image-cache on Android, the following error occurs when trying to get an image that does not exist at a url (basically, anything except a 200).

`Error: java.lang.AbstractMethodError: abstract method "void org.nativescript.widgets.Async$CompleteCallback.onError(java.lang.Object)"`

This seems to be a bug in the implementation.  If not, what might the problem be?

**Comment by tsonevn** [comment]
Hi @JWiseCoder,
I tested the described scenario with our [sample app](https://github.com/NativeScript/nativescript-sdk-examples-js/blob/master/app/ns-ui-widgets-category/image-cache/basics/basics-page.js#L22-L32) however was unable to recreate the issue. If I replace the URL of the image with another one, which does not have an image, the initially set placeholder will be loaded(I am replacing `https://avatars1.githubusercontent.com/u/7392261?v=4` with `https://github.com/NativeScript/NativeScript/issues`). 
Can you provide a sample project that can be used for reproducing the problem?

**Comment by JWiseCoder** [comment]
What about a different response code?  Like a 404?
And can the placeholder not be null?

I've been digging around in the code on this GitHub, and it seems to surround the implementation of image-cache for Android that creates a `org.nativescript.widgets.Async.CompleteCallback` callback.  It seems to set an onError handler, and I'm not sure what happens after that.  The error is coming from Android itself.

**Comment by JWiseCoder** [comment]
I'm having a different problem on the iOS side.  Should I make a separate issue for that?  (It involves the Cache object trying to set a nil value to NSCache)

**Comment by JWiseCoder** [comment]
I got the Android to work on some devices.  But it still fails on some.

**Comment by tsonevn** [comment]
Hi @JWiseCoder, 
I will check the case for iOS. Meanwhile, can you provide more info about the device, you are using while testing the issue? It will help if you can send us a sample project, that demonstrates how you are using the image-cache functionality and can be used for reproducing the problem.
Regarding issue #7036, in my opinion, the problem described in 7036 is similar to the Android one. On that matter, I am closing issue #7036 and would suggest keeping the discussion for both (Android and iOS cases) here.

**Comment by tsonevn** [comment]
@JWiseCoder,
I have just tested the image-cache with the app mentioned in my previous comment on iOS(iPhone X - iOS 12.1 and iPhone SE - iOS 12.1). However, I was unable to recreate the issue.  Regarding that, indeed it will help if you can send us a demo app, that can be used for recreating the iOS and Android problems.

**Comment by JWiseCoder** [comment]
Here is an example:

[https://play.nativescript.org/?template=play-ng&id=BxUcB8](https://play.nativescript.org/?template=play-ng&id=BxUcB8)

It crashes right away on my iPhone running iOS 12.1

**Comment by tsonevn** [comment]
Hi @JWiseCoder,
I checked your case while using the provided project on iPhone 6S and iPhone X devices with iOS 12.1 and on simulator iPhone Xr with iOS 12.1, however, was unable to recreate the error(`-[NSCache setObject:forKey:cost:]: attempt to insert nil value`) on my side. Are there any specific steps that are required for reproducing the issue? Meanwhile, try to clear up the project while deleting `node_modules`, `platforms`, `hooks` folders and `package-lock.json` file and then try to rebuild it.

Regarding Android, I was able to reproduce an error with the image-cache functionality and it seems to be related to NativeScript Angular.

**Comment by JWiseCoder** [comment]
I just ran it again on my XR and it crashed.  If I had to guess, I would say that the problem is with the try/catch in the httpRequest callback.  I think the source expects initWithData on UIImage will throw an exception if the data is empty or malformed, but in certain circumstances, this is not the case—it simply returns nil.  When that happens, it tries to add the nil image to the NSCache, and it fails.  But if an exception is thrown, it is caught and it runs the error handler instead.  Hence no NSCache attempt and no exception.  But that's just a guess.

And concerning Android, what is the solution?
